src/boat/private/fileManager.nim

```diff
@@ -15,36 +15,17 @@ import
   boatErrors,
   fileManagerUtils
 
-
-proc dir*(self: FileType): string =
-  ## returns the directory where different FileTypes are persisted
-  result = case self
-    of localManifest, captainsLog: cacheDir
-    else: tempDir
-
-proc path*(self: FileType, fname: string): string =
-  ## computes the filpath for a FileType
-  raise tddError
-  # result = self.dir / hash(fname)
-
-proc encode*[T: JsonNode | string](self: T): string =
-  ## encodes json & strings for saving to disk
-  raise tddError
-  # of JsonNode -> parse to string -> base64
-  # of string -> base64
-
-
 proc toDisk*[T: JsonNode | string | Config](
-  self: FileType,
-  fname: string,
+  ft: FileType,
+  use: string,
   data: T,
-  captainsLog: JsonNode
+  captain: JsonNode
   ): Future[string] {.async.} =
     ## persists data to cache or temp dir and returns path
     ## if file already exists, will overwrite if content is different
     ## any file saved to cacheDir will be added to the captains log
-    raise tddError
-    # fpath = self.path fname
+    let dirName = use.pathDir
```

Comment:
this should only be used for local and remoteManifest files and needs to be a in a case statement
- some/path/manifest.nim.ini and some/path should produce the same hash value

Reply:
its idiomatic nim to have a bunch of overloads, create a ticket to ticket for this